test/integration/http2_integration_test.cc

```diff
@@ -253,4 +255,193 @@ TEST_P(Http2IntegrationTest, SimultaneousRequestWithBufferLimits) {
   simultaneousRequest(1024 * 32, 1024 * 16);
 }
 
+Http2RingHashIntegrationTest::Http2RingHashIntegrationTest() {
+  config_helper_.addConfigModifier([&](envoy::api::v2::Bootstrap& bootstrap) -> void {
+    auto* cluster = bootstrap.mutable_static_resources()->mutable_clusters(0);
+    cluster->clear_hosts();
+    cluster->set_lb_policy(envoy::api::v2::Cluster_LbPolicy_RING_HASH);
+    for (int i = 0; i < num_upstreams_; i++) {
+      auto* socket = cluster->add_hosts()->mutable_socket_address();
+      socket->set_address(Network::Test::getLoopbackAddressString(version_));
+    }
+  });
+}
+
+Http2RingHashIntegrationTest::~Http2RingHashIntegrationTest() {
+  if (codec_client_) {
+    codec_client_->close();
+    codec_client_ = nullptr;
+  }
+  for (auto it = fake_upstream_connections_.begin(); it != fake_upstream_connections_.end(); ++it) {
+    (*it)->close();
+    (*it)->waitForDisconnect();
+  }
+}
+
+void Http2RingHashIntegrationTest::createUpstreams() {
+  for (int i = 0; i < num_upstreams_; i++) {
+    fake_upstreams_.emplace_back(new FakeUpstream(0, FakeHttpConnection::Type::HTTP1, version_));
```

Comment:
I think this would be a generally useful extension to the base integration test but definitely outside the scope of this PR.  Consider it as a follow up though :-)